engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -197,101 +266,97 @@ public void render(DefaultRenderingProcess.StereoRenderState stereoRenderState)
         switch (stereoRenderState) {
             case MONO:
                 currentRenderingStage = WorldRenderingStage.DEFAULT;
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_LEFT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_LEFT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_RIGHT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_RIGHT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = false;
                 break;
         }
 
-        if (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE) {
-            renderableWorld.updateAndQueueVisibleChunks();
-        } else {
-            renderableWorld.updateAndQueueVisibleChunks(false, false);
-        }
-
-        if (renderingConfig.isDynamicShadows()
-                && (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE)) {
-            DefaultRenderingProcess.getInstance().beginRenderSceneShadowMap();
-            renderShadowMap();
-            DefaultRenderingProcess.getInstance().endRenderSceneShadowMap();
-        }
+        renderableWorld.updateAndQueueVisibleChunks(isFirstRenderingStageForCurrentFrame);
 
-        DefaultRenderingProcess.getInstance().beginRenderReflectedScene();
-        glCullFace(GL11.GL_FRONT);
-        playerCamera.setReflected(true);
+        renderShadowMap();
         renderWorldReflection();
-        playerCamera.setReflected(false);
-        glCullFace(GL11.GL_BACK);
-        DefaultRenderingProcess.getInstance().endRenderReflectedScene();
 
-        renderWorld();
+        toggleWireFrameON(); // if wireframe is enabled
```

Comment:
I like the idea of grouping the wireframe toggle with at least the DefaultRenderingProcess.clear() line. On the other hand I'd rather not include the DefaultRenderingProcess.beginRenderSceneOpaque() because that would leave the matching line, DefaultRenderingProcess.endRenderSceneOpaque() unmatched.